core/entity/data/v1/v1.go

```diff
@@ -162,40 +162,63 @@ func (secret SecretStored) ToMapForK8s() map[string][]byte {
 
 	// If there is no template, use the secret’s value as is.
 	if secret.Meta.Template == "" {
-		err := json.Unmarshal(([]byte)(secret.Values[0]), &data)
-		if err != nil {
-			// need to reset data, if unmarshal failed
-			data = map[string][]byte{}
-			value := secret.Values[0]
-			data["VALUE"] = ([]byte)(value)
-		}
-
-		return data
+		return handleNoTemplate(data, secret.Values[0])
 	}
 
 	// Otherwise, apply the template.
 	newData, err := parseForK8sSecret(secret)
 	if err == nil {
-		data = make(map[string][]byte)
-		for k, v := range newData {
-			data[k] = ([]byte)(v)
-		}
-
-		return data
+		return convertMapToStringBytes(newData)
 	}
 
 	// If the template fails, use the secret’s value as is.
-	err = json.Unmarshal(([]byte)(secret.Values[0]), &data)
+	return handleTemplateFailure(data, secret.Values[0])
+}
+
+// handleNoTemplate is used when there is no template defined.
+// It attempts to unmarshal the 'value' string as JSON. If successful, it returns
+// a map with the JSON key-value pairs converted to []byte values; otherwise, it
+// returns a map with a single entry, "VALUE", containing the original 'value' as []byte.
+
+func handleNoTemplate(data map[string][]byte, value string) map[string][]byte {
+	var jsonData map[string]string
+	err := json.Unmarshal(([]byte)(value), &jsonData)
 	if err != nil {
-		// need to reset data if failed to unmarshal
-		data = map[string][]byte{}
```

Comment:
@farhan-pasha @abhishek44sharma — has this comment been resolved?

Also, there was a merge conflict; I fixed that on the branch, but please make sure things are still ok.

Cheers.

Reply:
Thanks for the review @abhishek44sharma 
I double checked it, in case of failure, unmarshal() returns an error and leaves the target struct unchanged. So I think we are good here. Please let me know if I missed something here I can go back and address it.